ros2cli/ros2cli/cli.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2016-2017 Dirk Thomas
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import signal
+
+from ros2cli.command import add_subparsers
+from ros2cli.command import get_command_extensions
+
+
+def main(*, script_name='ros2', argv=None, description=None, extension=None):
+    if description is None:
+        description = 'ros2 is an extensible command-line tool for ROS 2.'
```

Comment:
The argument allows other packages to pass a different `script_name` to this function (e.g. if you want an executable called `ros2topic`).

> and not use it here

I am not sure what you mean with this. The variable is used later.